Automattic/woocommerce-payments
# Fix AED and SAR currencies format

Fixes #6633 

#### Changes proposed in this Pull Request



This PR fixes the currency format for AED and SAR currencies. 



Previously, the AED and SAR currencies had the incorrect characters in use for the thousands and decimal separators. They were using the period for the thousands separator and the comma for the decimal separator. The correct format is the inverse of that. This PR only changes hardcoded values.





#### Testing instructions





* Check out this PR branch.
* All tests should pass.
* Create a product with a high price (ex. $1,234.56) if you don't have one already.
* Enable SAR and AED currencies in WooCommerce > Settings > Multi-currency.
* Add the multi-currency widget to your site's sidebar if not already there.
* Clear locale info and currency format transients: `npm run wp transient delete wcpay_locale_info && npm run wp transient delete wcpay_currency_format`
* Then comment out the AED and SAR entries in the WC core files:
  * AED locale: `/wp-content/plugins/woocommerce/i18n/locale-info.php:30-45`
  * SAR locale: `/wp-content/plugins/woocommerce/i18n/locale-info.php:3070-3085`
  * AED currency: `/wp-content/plugins/woocommerce/i18n/currency-info.php:129-132`
  * SAR currency: `/wp-content/plugins/woocommerce/i18n/currency-info.php:725-728`
* Navigate to the single product page that has the high price.
* Switch your currency to AED and observe the price. The format should have a comma thousands separator and a period decimal separator. 
* Switch your currency to SAR and observe the price. The format should have a comma thousands separator and a period decimal separator.

Example screenshots with the correct thousands and decimal separators:

#### AED

![Screenshot on 2023-08-28 at 15_08_00](https://github.com/Automattic/woocommerce-payments/assets/60988591/74e0842c-0f68-466c-affd-3ec3027e28ab)

#### SAR

![Screenshot on 2023-08-28 at 15_07_30](https://github.com/Automattic/woocommerce-payments/assets/60988591/535c9835-d321-49c0-b8ba-8928f997832b)

-------------------


**Post merge**
